### PR TITLE
ESM samples - update to 4.25 and update deps

### PR DESCRIPTION
**Pull request by andygup**
Notable items:
- Custom widget sample added `vite.config.ts`
- Custom widget sample staying at Vite 2 pending further investigation of an issue with data binding at Vite 3. Will update sample at a future date.
- Custom worker sample updated to rollup 3 and refactored to replace `graphic.clone()` due to bug.
- Custom worker sample replaced old terser plugin with `@rollup/plugin-terser`, and added a degraded build performance note in the readme related to the new plugin. More info [here](https://github.com/rollup/plugins/issues/1334), hopefully there will be a plugin update soon.
- Vue sample updated to Vite 3
- Create React App sample updated to use Vite and renamed via https://github.com/Esri/jsapi-resources/commit/97d8aae45938f73bd35f5b36ee854b36e0833e7f, also old sample is available under the [Legacy tag](https://github.com/Esri/jsapi-resources/releases/tag/legacy).
- Removed Ember sample, it's now available under the [Legacy tag](https://github.com/Esri/jsapi-resources/releases/tag/legacy) and documented in the esm-samples README.
- Will update Angular sample to v15 when it comes available (currently at rc2).

### CODE REVIEW

File: esm-samples/README.md [hunk 1]
```diff
@@ -8,15 +8,19 @@ Run `npm install` in a project directory and then start adding ArcGIS API for Ja
...
 
 For each release, we track certain metrics including bundle size and web application performance. These files can be found in the [/esm-samples/.metrics](./.metrics/) directory. You can read more about how we capture this information this [README](../.github/scripts/README.md).
 
-Bundle size is dependant on multiple factors and will vary based on framework, module bundler, transpiling and related configurations, as well as the ArcGIS API for JavaScript functionality that has been implemented. The compile process can output several hundred or more bundles on-disk due to using dynamic imports. At runtime, your application will typically only load a portion of the bundles depending on the `@arcgis/core` functionality used. 
+Bundle size is dependant on multiple factors and will vary based on framework, module bundler, transpiling and related configurations, as well as the ArcGIS API for JavaScript functionality that has been implemented. The compile process can output several hundred or more bundles on-disk due to using dynamic imports. At runtime, your application will typically only load a portion of the bundles depending on what `@arcgis/core` modules are used in your project.
```
Comment (dasa) [inline esm-samples/README.md:19]: How about one of these?
- depending on the `@arcgis/core` modules used in your project.
- depending on which `@arcgis/core` modules are used in your project.

File: esm-samples/jsapi-custom-workers/README.md [hunk 2]
```diff
@@ -13,7 +14,7 @@ The key to using custom workers is building the workers separately from your mai
 // rollup.worker.config.js
 import commonjs from "@rollup/plugin-commonjs";
 import resolve from "@rollup/plugin-node-resolve";
-import { terser } from "rollup-plugin-terser";
+import terser from '@rollup/plugin-terser';
```
Comment (dasa) [inline esm-samples/jsapi-custom-workers/README.md:17]: Use " for consistency

File: esm-samples/jsapi-vue/package.json [hunk 3]
```diff
@@ -7,11 +7,11 @@
     "preview": "vite preview"
   },
   "dependencies": {
-    "@arcgis/core": "~4.24.0",
-    "vue": "^3.2.37"
+    "@arcgis/core": "^4.25.0",
```
Comment (dasa) [inline esm-samples/jsapi-vue/package.json:10]: Did you mean to change these from ~ to ^?

**Review by dasa — CHANGES_REQUESTED** [review]

File: esm-samples/README.md (hunk 1 above)
Comment (andygup) [reply esm-samples/README.md]: Done, I went with "which".

File: esm-samples/jsapi-custom-workers/README.md (hunk 2 above)
Comment (andygup) [reply esm-samples/jsapi-custom-workers/README.md:17]: Fixed.

File: esm-samples/jsapi-vue/package.json (hunk 3 above)
Comment (andygup) [reply esm-samples/jsapi-vue/package.json:10]: Nope, that was an accident. Fixed.

Comment (dasa) [reply esm-samples/jsapi-vue/package.json:10]: There's still three places using "^4.25.0"

File: esm-samples/README.md [hunk 4]
```diff
@@ -8,15 +8,19 @@ Run `npm install` in a project directory and then start adding ArcGIS API for Ja
...
+If you run the performance metrics script locally, the performance indicators will vary depending on the device or virtual machine that runs the script. The metrics are best used for analysing trends over time, and they should always be run on the same device in order to keep as many of the variables constant as possible. The calculations are dependant on many hardware and software factors including CPU load, memory usage and internet speed during the time window that the script is running. For example, the numbers you see using a GitHub action may be significantly different from when you run the scripts locally on your development machine.
 
-### TypeScript
+### Archived Samples
```
Comment (odoe) [inline esm-samples/README.md:23]: Do we want to mention the create-react-app sample?

**Review by odoe — APPROVED** [review]
Looks good, one comment

File: esm-samples/README.md (hunk 4 above)
Comment (andygup) [reply esm-samples/README.md:23]: Good catch. Done.

File: esm-samples/jsapi-vue/package.json (hunk 3 above)
Comment (andygup) [reply esm-samples/jsapi-vue/package.json]: Argh, relearning how to use project search, lol, I found and fixed.

**Review by dasa — APPROVED** [review]
👍🏻

**Pull request merged**